pkg/histogram/histogram_test.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package histogram
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestHistogram_NewIntervalsFromExp2Slots(t *testing.T) {
+	t.Parallel()
+
+	const unit = UnitMicroseconds
+
+	testTable := []struct {
+		description string
+		slots       []uint32
+		expected    *Histogram
+	}{
+		{
+			description: "Empty slots",
+			slots:       []uint32{},
+			expected: &Histogram{
+				Unit:      unit,
+				Intervals: nil,
+			},
+		},
+		{
+			description: "With 1 slot",
+			slots:       []uint32{1},
```

Comment:
nit: I'd use higher numbers than 1 in this case, to avoid those being messed up with indexes. (8888, 9999, ...)